src/shogun/features/CombinedDotFeatures.cpp

```diff
@@ -129,67 +134,53 @@ float64_t CCombinedDotFeatures::dense_dot(int32_t vec_idx1, const float64_t* vec
 
 void CCombinedDotFeatures::dense_dot_range(float64_t* output, int32_t start, int32_t stop, float64_t* alphas, float64_t* vec, int32_t dim, float64_t b) const
 {
-	if (stop<=start)
-		return;
+	ASSERT(stop > start)
 	ASSERT(dim==num_dimensions)
 
 	uint32_t offs=0;
-	bool first=true;
 	int32_t num=stop-start;
-	float64_t* tmp=SG_MALLOC(float64_t, num);
-
+	
+	SGVector<float64_t> tmp(num);
```

Comment:
`tmp.zero()` reads a bit easier
or you can also use linalg.
But I guess std::fill is also fine ... just saying :)

Reply:
I used fill with `output`, which is a raw pointer. can't use `linalg` or `zero`